platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -245,6 +257,51 @@ public Object getAttribute(String name) {
             return doPaosRequest(request, response);
         }
 
+        if (userAgentCheck && userAgentIsNotBrowser(httpRequest)) {
+            SecurityLogger.audit("Attempting to log client in as a legacy system.");
+            //if we get here, it is most likely an older DDF that is federating
+            //it isn't going to understand the redirect to the IdP and it doesn't support ECP
+            //so we need to fall back to other handlers to allow it to log in using PKI, Basic or Guest
```

Comment:
You don't configure the IdP with any other handlers, which is why I went this route. It is an interesting thought though. I could make it fall through to any handlers that might come after the IdpHandler, but the system would also need to be configured that way in order for any legacy clients to fall back appropriately. Also, this is for legacy clients, so while it is possible for some third party to deploy their own authentication mechanism, I don't expect them to need that to work with our IdP. I'd expect them to just change the auth method over to their custom one.